src/liballoc/boxed.rs

```diff
@@ -414,3 +416,69 @@ impl<'a,A,R> FnOnce<A> for Box<FnBox<A,Output=R>+Send+'a> {
 }
 
 impl<T: ?Sized+Unsize<U>, U: ?Sized> CoerceUnsized<Box<U>> for Box<T> {}
+
+impl<T: Clone> Clone for Box<[T]> {
+    fn clone(&self) -> Self {
+        let mut alloc = unsafe {
+            heap::allocate(mem::size_of::<T>() * self.len(),
```

Comment:
You can't do this unconditionally; you need to special case `mem::size_of::<T> = 0` to not allocate (use alloc::heap::EMPTY).